File: stripe/src/test/java/com/stripe/android/model/PaymentIntentRedirectDataTest.java
package com.stripe.android.model;

import android.net.Uri;

import java.util.HashMap;
import java.util.Map;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertNull;

@RunWith(RobolectricTestRunner.class)
public class PaymentIntentRedirectDataTest {

    @Test
    public void create_withBothFieldsPopulated_shouldReturnCorrectObject() {
        final String url = "https://example.com";
        final String returnUrl = "yourapp://post-authentication-return-url";
        final Map<String, String> redirectMap = new HashMap<>();
        redirectMap.put(StripeIntent.RedirectData.FIELD_URL, url);
        redirectMap.put(StripeIntent.RedirectData.FIELD_RETURN_URL, returnUrl);

        final PaymentIntent.RedirectData redirectData =
                StripeIntent.RedirectData.create(redirectMap);
        assertNotNull(redirectData);
        assertEquals(Uri.parse(url), redirectData.url);
        assertEquals(returnUrl, redirectData.returnUrl);
    }

    @Test
    public void create_withOnlyUrlFieldPopulated_shouldReturnCorrectObject() {
        final String url = "https://example.com";
        final Map<String, String> redirectMap = new HashMap<>();
        redirectMap.put(StripeIntent.RedirectData.FIELD_URL, url);

        final PaymentIntent.RedirectData redirectData =
                StripeIntent.RedirectData.create(redirectMap);
        assertNotNull(redirectData);
        assertEquals(Uri.parse(url), redirectData.url);
        assertNull(redirectData.returnUrl);
    }

    @Test
    public void create_withInvalidData_shouldReturnNull() {
        assertNull(StripeIntent.RedirectData.create(new HashMap<>()));
    }
}

File: stripe/src/test/java/com/stripe/android/model/PaymentIntentRedirectDataTest.kt
package com.stripe.android.model

import android.net.Uri
import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertNotNull
import kotlin.test.assertNull
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner

@RunWith(RobolectricTestRunner::class)
class PaymentIntentRedirectDataTest {

    @Test
    fun create_withBothFieldsPopulated_shouldReturnCorrectObject() {
        val url = "https://example.com"
        val returnUrl = "yourapp://post-authentication-return-url"
        val redirectMap = mapOf(
            StripeIntent.RedirectData.FIELD_URL to url,
            StripeIntent.RedirectData.FIELD_RETURN_URL to returnUrl
        )

        val redirectData =
            requireNotNull(StripeIntent.RedirectData.create(redirectMap))
        assertEquals(Uri.parse(url), redirectData.url)
        assertEquals(returnUrl, redirectData.returnUrl)
    }

    @Test
    fun create_withOnlyUrlFieldPopulated_shouldReturnCorrectObject() {
        val url = "https://example.com"
        val redirectMap = mapOf(StripeIntent.RedirectData.FIELD_URL to url)

        val redirectData = StripeIntent.RedirectData.create(redirectMap)
        assertNotNull(redirectData)
        assertEquals(Uri.parse(url), redirectData.url)
        assertNull(redirectData.returnUrl)
    }

    @Test
    fun create_withInvalidData_shouldReturnNull() {
        assertNull(StripeIntent.RedirectData.create(emptyMap<String, String>()))
    }
}
